cli/tests/pcluster/models/test_cluster_stack.py

```diff
@@ -110,24 +111,30 @@ def test_wait_for_stack_update(mocker, stack_statuses):
 def test_update_stack_template(mocker, error_message):
```

Comment:
Adding this here but it is more of a general comment not only related to your changes:
some of these unit tests are getting very complex and it's hard to understand what we are testing and what not. also if we write unit tests that are more complex than the code they are testing we end-up in a situation where we need to test the tests. We are also doing a lot of mocking which probably is necessary but at the same time it makes very hard to understand what is the real code under test. Usually when a lot of mocking is required this is a symptom that the function under test is complex. Also we could consider having multiple simple test function if a single parametrized one risks to become too complex.